x-pack/plugins/infra/public/pages/metrics/hosts/helpers.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BooleanRelation, buildCombinedFilter, buildPhraseFilter } from '@kbn/es-query';
+import type { DataView } from '@kbn/data-views-plugin/common';
+
+export const buildCombinedHostsFilter = ({
```

Comment:
Cool that you're using `es-query` utils functions :). We already have [this function `createHostsFilter`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/pages/metrics/hosts/utils.ts#L10) that does exactly the same, but manually, which might not be so bad because of the dependency that this approach has on the `dataview`. 

Could we adapt one of those functions in order to keep only one for this?

Reply:
I can't use this filter directly because it uses different types (phrases not combined) and the result is slightly different (`one of` instead of a `combined` filter with `OR`) But I can try to move the method there and change it a bit 👍 
<img width="970" alt="image" src="https://github.com/elastic/kibana/assets/14139027/c2fbef7c-2868-4de7-8a48-a647dd898dd8">

Reply:
Refactored ✅